sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1241,27 +1219,24 @@ trait BasicWindowCalc {
    */
   def computeBasicWindow(cb: ColumnarBatch): Array[cudf.ColumnVector] = {
     closeOnExcept(new Array[cudf.ColumnVector](boundWindowOps.length)) { outputColumns =>
-      val inputSpillable = SpillableColumnarBatch(
-        GpuProjectExec.project(cb, initialProjections),
-        SpillPriorities.ACTIVE_BATCHING_PRIORITY)
-
-      // this takes ownership of `inputSpillable`
-      aggregations.doAggsAndClose(
-        isRunningBatched,
-        boundOrderSpec,
-        orderByPositions,
-        partByPositions,
-        inputSpillable,
-        outputColumns)
-
-      // if the window aggregates were successful, lets splice the passThrough
-      // columns
+      // First the pass through unchanged columns
       passThrough.foreach {
         case (inputIndex, outputIndex) =>
           outputColumns(outputIndex) =
             cb.column(inputIndex).asInstanceOf[GpuColumnVector].getBase.incRefCount()
       }
 
+      // make spillable
+      val spillableBatch = SpillableColumnarBatch(
+          GpuProjectExec.project(cb, initialProjections),
+          SpillPriorities.ACTIVE_BATCHING_PRIORITY
+        )
+
+      withRetryNoSplit(spillableBatch) { initProjCb =>
```

Comment:
I thought that we were going to have a `computeBasicWindow` that did no retry at all, and then we could wrap all of it in a retry block that would do the retry.  That way computeRunning could wrap it without us worrying about nested retries and trying to figure out how to deal with the pass through columns and the fact that they are not spillable.

Reply:
This is in place now, but there is no version that does a retry now and I may still need to add that back in for other call sites.